src/hooks/useWalletConnection.ts

```diff
@@ -189,17 +205,41 @@ export const useWalletConnection = () => {
         walletConnectionType: walletConnection?.type,
       };
     },
-    [isConnectedGraz, signerGraz, isConnectedWagmi, signerWagmi, ready, authenticated, login]
+    [
+      walletConnectionType,
+      isConnectedWagmi,
+      ready,
+      authenticated,
+      login,
+      isConnectedGraz,
+      stringGetter,
+      connectGraz,
+      saveEvmAddress,
+      connectPhantom,
+      connectWagmi,
+      walletConnectConfig,
+    ]
   );
 
   const disconnectWallet = useCallback(async () => {
     saveEvmAddress(undefined);
     saveDydxAddress(undefined);
+    saveSolAddress(undefined);
 
     if (isConnectedWagmi) await disconnectWagmi();
```

Comment:
I think we may have to also call the disconnect provider method for Phantom. I noticed that when I tried to disconnect Phantom, I was unable to reconnect without refreshing the tab.

Reply:
✅